ingress-controller/test/integration/plugin_test.go

```diff
@@ -802,14 +804,27 @@ func TestPluginNullInConfig(t *testing.T) {
 		require.NoError(t, err, "failed to create Kong client")
 		plugins, err := kc.Plugins.ListAll(ctx)
 		require.NoError(t, err, "failed to list plugins")
-		if len(plugins) != 1 {
-			return false
-		}
-		plugin := plugins[0]
-		if plugin.Name == nil || *plugin.Name != "datadog" {
+
+		datadogPlugin, found := lo.Find(plugins, func(p *kong.Plugin) bool {
+			return p.Name != nil && *p.Name == "datadog"
+		})
+		if !found {
+			t.Logf("datadog plugin not found. %d Plugins found: %s",
+				len(plugins),
+				strings.Join(
+					lo.Map(plugins, func(p *kong.Plugin, _ int) string {
+						return lo.FromPtrOr(p.Name, "_")
+					}),
+					","),
+			)
 			return false
 		}
-		configPrefix, ok := plugin.Config["prefix"]
+
+		configJSON, err := json.Marshal(datadogPlugin.Config)
+		require.NoError(t, err)
```

Comment:
Since this is in `require.Eventually`, this position can `return false` instead of calling `require.NoError`, which would cause it to exit early.